ginepro/src/balanced_channel.rs

```diff
@@ -188,4 +192,42 @@ impl<T: LookupService + Send + Sync + 'static + Sized> LoadBalancedChannelBuilde
 
         LoadBalancedChannel(channel)
     }
+
+    /// Tries to resolve the set of IPs from the (Hostname)[`ServiceDefinition::hostname`]
+    /// within the given timeout before creating a [`LoadBalancedChannel`].
+    ///
+    /// If no timeout is specified it will default to 2 seconds.
+    pub async fn build_and_resolve<O: Into<Option<Duration>>>(
```

Comment:
More than a separate method I'd suggest going for a configuration parameter (`resolve_eagerly: bool`) with its method on the builder.
We should also try to reuse most of the logic in the existing `channel` method instead of duplicating it.

Reply:
So you suggest something like:
```
pub async fn channel(&self) -> Result<LoadBalancedChannel, anyhow::Error> {
   if self.resolve_eagerly {
        self.lookup_service.resolve()?;
   }
}
```

If so the concern I have is that we impose the following to the user:
* requiring the function to be async in all scenarios even if it doesn't necessarily need to.
* the function always returns a result even if `resolve_eagerly` is set to `false` in which case the function would never error.